wcfsetup/install/files/lib/acp/form/BoxEditForm.class.php

```diff
@@ -239,6 +257,26 @@ public function readData()
             $this->invertPermissions = $this->box->invertPermissions;
 
             $this->readBoxImages();
+
+            $conditions = $this->box->getVisibilityConditions();
+            $conditionsByObjectTypeID = [];
+            foreach ($conditions as $condition) {
+                $conditionsByObjectTypeID[$condition->objectTypeID] = $condition;
+            }
+
+            foreach ($this->groupedConditionObjectTypes as $objectTypes1) {
```

Comment:
This overlaps with the logic in line 170+ above. How about introducing a helper method to do this, it will also save you from the awkward names like `$objectTypes2`:
```php
private function toFlatList(array $groupedObjects): array { /* … */ }
```

Reply:
As a note, this is fully c&p code. However, since you are the second to note this and find the code hard to read, I have adjusted this and introduced the appropriate helper method. It may make sense to make the method globally available in the future and to use it everywhere where the code is used.